src/main/java/com/capgemini/data/ReservationRepository.java

```diff
@@ -1,18 +1,36 @@
 package com.capgemini.data;
 
+import com.capgemini.domain.Guest;
 import com.capgemini.domain.Reservation;
-import java.util.ArrayList;
-import java.util.List;
+import com.capgemini.domain.Room;
+import com.capgemini.domain.RoomType;
+import org.springframework.stereotype.Component;
 
+import java.util.*;
+
+@Component
 public class ReservationRepository {
 
-    private List<Reservation> reservations = new ArrayList<>();
+    private List<Reservation> reservations;
+
+    public ReservationRepository(){
+        reservations = new ArrayList<>();
+    }
 
     public void addReservation(Reservation reservation) {
         reservation.setReservationID((reservations.size() + 1));
         reservations.add(reservation);
     }
 
+    public Reservation getReservationById(int id){
+        return getAllReservations().stream().filter(x -> x.getReservationID() == id).findFirst().orElse(null);
+    }
+
+    public void updateReservation(int id, Reservation reservation){
+        Reservation replace = getReservationById(id);
+        reservations.remove(replace);
+        reservations.add(reservation);
```

Comment:
Dit zou eigenlijk reservations.set(index, reservation) moeten zijn in plaats van remove, add... (Sorry is mijn eigen code)